Set/Models/GameModel.swift

```diff
@@ -122,15 +126,37 @@ struct GameModel {
         }
     }
     
-    mutating private func getBonus(for player: Player) {
+    private mutating func isFasterThanPreviousMatch() -> Bool {
+        let matchDate = Date.now
+        let matchTimeInterval = matchDate.timeIntervalSince(previousMatchDate ?? startGameDate)
+        
+        guard let previousMatchTimeInterval = previousMatchTimeInterval else {
+            self.previousMatchTimeInterval = matchTimeInterval
+            previousMatchDate = matchDate
+            return false
+        }
+        
+        self.previousMatchTimeInterval = matchTimeInterval
+        previousMatchDate = matchDate
+        
+        return matchTimeInterval < previousMatchTimeInterval
+    }
+    
+    private mutating func getBonus(for player: Player) {
```

Comment:
I would make `func bonus(for turnDutaton: TimeInteval) -> Int` that do just one thing, takes duration and return bonus >= 0